core/src/main/java/org/elasticsearch/env/Environment.java

```diff
@@ -179,6 +180,52 @@ public Path resolveRepoFile(String location) {
     }
 
     /**
+     * Checks if the specified URL is pointing to the local file system and if it does, resolves the specified url
+     * against the list of configured repository roots
+     *
+     * If the specified url doesn't match any of the roots, returns null.
+     */
+    public URL resolveRepoURL(URL url) {
+        try {
+            if ("file".equalsIgnoreCase(url.getProtocol())) {
+                if (url.getHost() == null || "".equals(url.getHost())) {
+                    // only local file urls are supported
```

Comment:
this looks good. i forgot how crazy URL was here with null versus empty strings for different pieces...